packages/flutter/lib/src/gestures/force_press.dart

```diff
@@ -215,8 +215,18 @@ class ForcePressGestureRecognizer extends OneSequenceGestureRecognizer {
     assert(_state != _ForceState.ready);
     // A static pointer with changes in pressure creates PointerMoveEvent events.
     if (event is PointerMoveEvent || event is PointerDownEvent) {
-      final double pressure = interpolation(event.pressureMin, event.pressureMax, event.pressure);
-      assert(pressure.isNaN ? true : (pressure <= 1.0 && pressure >= 0.0));
+      double pressure = interpolation(event.pressureMin, event.pressureMax, event.pressure);
+      assert(
+        event.pressureMin < 0.0 ||
+        event.pressureMax > 0.0 ||
+        pressure.isNaN ||
+        (pressure < 1.0 && pressure > 0.0)
+      );
+      // If the device incorrectly reports a pressure outside of pressureMin
+      // and pressureMax, we still want this recognizer to respond normally.
+      if (!pressure.isNaN)
+        pressure = pressure.clamp(0.0, 1.0);
```

Comment:
Wouldn't the assert above fire if pressure is not within [0.0, 1.0]? What's the purpose of doing both?

Reply:
The assert reports true if the pressure is greater than the device reported max pressure. The clamp fixes this case.

Reply:
but if the assert fires, we never get here

Reply:
(i think the assert won't fire, though.)